src/main/kotlin/app/nush/bot/commands/Nick.kt

```diff
@@ -0,0 +1,96 @@
+package app.nush.bot.commands
+
+import app.nush.bot.Config.Companion.config
+import com.jessecorbett.diskord.api.rest.CreateDM
+import com.jessecorbett.diskord.dsl.Bot
+import com.jessecorbett.diskord.dsl.CommandSet
+import com.jessecorbett.diskord.dsl.command
+import com.jessecorbett.diskord.util.authorId
+import com.jessecorbett.diskord.util.changeNickname
+import com.jessecorbett.diskord.util.sendMessage
+import com.jessecorbett.diskord.util.words
+
+object Nick : Command {
+    lateinit var bot: Bot
+    override fun init(bot: Bot, prefix: CommandSet) {
+        this.bot = bot
+        with(prefix) {
+            command("nick") {
+                request(
+                    bot,
+                    authorId,
+                    words.drop(1).joinToString(separator = " ")
+                )
+            }
+        }
+    }
+
+    suspend fun request(bot: Bot, userId: String, newName: String) {
+        with(bot) {
+            val guildClient = clientStore.guilds[config.guildId]
+            val channelClient = clientStore.channels[config.excoChannelId]
+            clientStore.channels[clientStore.discord.createDM(
+                CreateDM(
+                    userId
+                )
+            ).id].sendMessage("Your rename request has been sent to the exco, it will be processed as soon as possible.")
+            val renameMessage =
+                channelClient.sendMessage("${guildClient.getMember(userId).nickname}(<@!$userId>) has requested to be renamed to \"$newName\". ✅ if you want to accept the rename request, ❎ if you do not want to accept the rename request")
+            renameMessage.react("✅")
+            renameMessage.react("❎")
+            reactionAdded { messageReaction ->
+                try {
+                    if (!guildClient.getMember(messageReaction.userId).user?.isBot!!) {
+                        val newName: String = channelClient.getMessage(messageReaction.messageId).content.subSequence(
+                            channelClient.getMessage(messageReaction.messageId).content.indexOf("\"") + 1,
```

Comment:
You can use a variable to store `channelClient.getMessage(messageReaction.messageId)` to prevent repeated calls. Try using the intellij shortcut <kbd>Ctrl</kbd> + <kbd>Alt</kbd> + <kbd>v</kbd>
https://www.jetbrains.com/help/idea/extract-variable.html